### DISCUSSION TITLE
Updating a variable in JS but pyscript keeps using the old value

### DISCUSSION THREAD

**Discussion by ghost** [pyscript/pyscript#1376] [Q&A]
Hello there! 

I don't know if it's intended, but when I update a value in JS, and then call a function from pyscript that uses that variable, it keeps using the initial value. For example: https://codepen.io/clodoaldofavero/pen/dygogLW 

Is there any way to get the updated variable value?

**Comment by hoodmane** [comment]
Yes this is the intended behavior. It's pretty much the same as if you had a Python module called `example.py`:
```py
test_variable = 5
def inc():
  global test_variable
  test_variable += 1
```
and then you did:
```py
from example import test_variable, inc
import example

assert test_variable == 5
assert example.test_variable == 5
inc()
assert test_variable == 5 # still 5
assert example.test_variable == 6 # this is 6 now
```

**Reply by WebReflection** [reply]
FWIWI this might not be surprising in Python but it *might* be somehow surprising in JS:

```js
// module.js
export let value = 0;
export const inc = () => {
  value++;
};

// consumer
import {value, inc} from './module.js';

console.assert(value === 0);
inc();
console.assert(value === 1);
```

Maybe something we might improve via related documentation?
